File: 71A.py
"""
Way Too Long Words

Sometimes some words like "localization" or "internationalization"
are so long that writing them many times in one text is quite tiresome.
"""

n = int(input(""))
for i in range(n):
    word = input("")
    if len(word)>10:
        print(word[0] + str(len(word)-2) + word[-1])
    else:
        print(word)

File: 71A.cpp
/**
Way Too Long Words

Sometimes some words like "localization" or "internationalization"
are so long that writing them many times in one text is quite tiresome.
*/




#include <iostream>
#include <string>

using namespace std;

int main()
{
    int n, len;
    string word;
    cin>>n;

    for(int i=0 ; i<n ; ++i)
    {
        cin>>word;
        len = word.size();
        if (len>10)
        {
            printf("%c%d%c\n", word[0], len-2, word[len-1]);
        }
        else
        {
            cout<<word<<"\n";
        }
    }
    
    return 0;
}